src/overlays/actors/ovl_Obj_Switch/z_obj_switch.c

```diff
@@ -28,183 +82,961 @@ const ActorInit Obj_Switch_InitVars = {
+                                          this->dyna.actor.world.pos.z, &this->dyna.actor.shape.rot);
+    Matrix_Scale(this->dyna.actor.scale.x, this->dyna.actor.scale.y, this->dyna.actor.scale.z, MTXMODE_APPLY);
+    Collider_UpdateSpheres(0, &this->colliderJntSph);
+}
+
+void ObjSwitch_InitTrisCollider(ObjSwitch* this, GlobalContext* globalCtx, ColliderTrisInit* init) {
+    s32 pad;
+    s32 i;
+    s32 j;
+    Vec3f vtx[3];
+
+    Collider_InitTris(globalCtx, &this->colliderTris);
+    Collider_SetTris(globalCtx, &this->colliderTris, &this->dyna.actor, init, this->colliderTrisElements);
+    Matrix_StatePush();
+    Matrix_SetStateRotationAndTranslation(this->dyna.actor.world.pos.x,
+                                          this->dyna.actor.world.pos.y +
+                                              this->dyna.actor.shape.yOffset * this->dyna.actor.scale.y,
+                                          this->dyna.actor.world.pos.z, &this->dyna.actor.shape.rot);
+
+    for (i = 0; i < ARRAY_COUNT(this->colliderTrisElements); i++) {
+        if (this) {}
+        for (j = 0; j < ARRAY_COUNT(vtx); j++) {
+            Matrix_MultiplyVector3fByState(&init->elements[i].dim.vtx[j], &vtx[j]);
+        }
+        Collider_SetTrisVertices(&this->colliderTris, i, &vtx[0], &vtx[1], &vtx[2]);
+    }
+    Matrix_StatePop();
+}
+
+Actor* ObjSwitch_SpawnIce(ObjSwitch* this, GlobalContext* globalCtx) {
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->dyna.actor, globalCtx, ACTOR_OBJ_ICE_POLY,
+                       this->dyna.actor.world.pos.x, this->dyna.actor.world.pos.y - 25.0f, this->dyna.actor.world.pos.z,
+                       this->dyna.actor.world.rot.x, this->dyna.actor.world.rot.y, this->dyna.actor.world.rot.z,
+                       0xFF32);
+}
+
+void ObjSwitch_SetSwitchFlagState(ObjSwitch* this, GlobalContext* globalCtx, s32 setFlag) {
+    if (setFlag) {
+        s32 flag = OBJ_SWITCH_GET_SWITCH_FLAG(&this->dyna.actor);
+
+        Flags_SetSwitch(globalCtx, flag);
+    } else {
+        s32 flag = OBJ_SWITCH_GET_SWITCH_FLAG(&this->dyna.actor);
+
+        Flags_UnsetSwitch(globalCtx, flag);
+    }
+}
+
+void ObjSwitch_CrystalUpdateTimer(ObjSwitch* this) {
+    this->crystalAnimTimer++;
+}
+
+void func_8093AE88(ObjSwitch* this) {
+    if (this->unk_172 != 0) {
+        ActorCutscene_Stop(this->dyna.actor.cutscene);
+        this->unk_172 = 0;
+    }
+}
+
+void ObjSwitch_PlayFootSwitchSfx(ObjSwitch* this) {
+    if (this->unk_165 <= 0) {
+        Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_FOOT_SWITCH);
+    }
+}
+
+void ObjSwitch_PlayDiamondSwitchSfx(ObjSwitch* this) {
+    if (this->unk_165 <= 0) {
+        Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_DIAMOND_SWITCH);
+    }
+}
+
+void func_8093AF1C(ObjSwitch* this, s32 arg1) {
+    if (DynaPolyActor_IsInRidingMovingState(&this->dyna)) {
+        this->unk_170 = arg1;
+    }
+}
+
+void func_8093AF54(ObjSwitch* this, GlobalContext* globalCtx) {
```

Comment:
This looks like it's rotating during a cutscene.
```suggestion
void ObjSwitch_TurnPlayer(ObjSwitch* this, GlobalContext* globalCtx) {
```

Reply:
Code doesn't update any rotation variables, so I don't know where you got that idea. I ran a sim on it and it seems like it tries to snap the player to the boundaries of a square that's 48x48/96x96 depending on if it's a regular or big floor switch